src/accounts/ethereum.ts

```diff
@@ -145,10 +145,15 @@ export function NewAccount(derivationPath = "m/44'/60'/0'/0/0"): { account: ETHA
  * Get an account from a Web3 provider (ex: Metamask)
  *
  * @param  {ethers.providers.ExternalProvider} provider
+ * @param requestedRpc Use this params to change the RPC endpoint;
  */
-export async function GetAccountFromProvider(provider: ethers.providers.ExternalProvider): Promise<ETHAccount> {
+export async function GetAccountFromProvider(
+    provider: ethers.providers.ExternalProvider,
+    requestedRpc: ChangeRpcParam = RpcChainType.ETH,
+): Promise<ETHAccount> {
     const ETHprovider = new ethers.providers.Web3Provider(provider);
     const jrw = new JsonRPCWallet(ETHprovider);
+    await jrw.changeNetwork(requestedRpc);
```

Comment:
Can you set this as an optional argument instead, and not call the `changeNetwork` method unless `requestedRpc` is set.

Reply:
The purpose was to switch the user to the suitable protocol automatically,
On metamask if the user is already on the right one, this call will not open any UI, and the user wont see any difference

Do you think this change is still required regarding this information?